dbt/include/global_project/macros/materializations/models/incremental/merge.sql

```diff
@@ -22,7 +22,7 @@
             {% endfor %}
         {% else %}
             {% set unique_key_match %}
-                DBT_INTERNAL_SOURCE.{{ unique_key }} = DBT_INTERNAL_DEST.{{ unique_key }}
+                {{ adapter.dispatch('equals', 'dbt')(DBT_INTERNAL_SOURCE.{{ unique_key }}, DBT_INTERNAL_DEST.{{ unique_key }}) }}
```

Comment:
`adapter.dispatch('equals', 'dbt')` should just be `equals`. The dispatch happens within the `equals` macro, which then dispatches to `default__equals` or the platform-specific version, e.g. `postgres__equals`.

I would also test this against a concrete adapter, like `dbt-postgres`. I don't know if you can do the double expansion (`{{}}` inside another `{{}}`).